lib/util/http/host-rules.ts

```diff
@@ -120,24 +120,30 @@ export function applyHostRules<GotOptions extends HostRulesGotOptions>(
   const options: GotOptions = { ...inOptions };
   const foundRules = findMatchingRules(options, url);
   const { username, password, token, enabled, authType } = foundRules;
+  const { host } = new URL(url);
```

Comment:
this throws for invalid url.

Reply:
I'm not sure it can be invalid by this point - will check

Reply:
Added try/catch